### ISSUE TITLE
Product image upload error

### ISSUE THREAD

**Issue by LucaGallinari** [magespecialist/m1-MSP_DevTools#12]
It's very strange. I am not on a fresh project this time and i have some third parties module installed. This issue happen when i try to upload one or more images for a product (new or existent). I know that is this module that produce the error because if i disable it the uploader will start to operate again. After i click on "upload images" the upload fail and in the console log is logged the resulting JSON response, as this:
![schermata del 2016-09-14 12-32-41](https://cloud.githubusercontent.com/assets/5822554/18509381/954e16f8-7a78-11e6-9c32-e4aaecaa2b85.png)

As you can see there is a valid JSON response until the red line that i draw and then there is your module "script" tag with the JS code inside. Everything inside a JSON response.

**Comment by phoenix128** [comment]
It probably depends on `MSP_DevTools_Helper_Data::isAjax` method and it probably depends on client not correctly sending `x-requested-with` header.

We are testing it on our dev environment and we will give you a feedback ASAP.

**Comment by phoenix128** [comment]
Ok, you have to replace `MSP_DevTools_Helper_Data::isAjax` method with:

```
    public function isAjax()
    {
        if (is_null($this->_isAjax)) {
            $requestWith = strtolower(Mage::app()->getRequest()->getHeader('x-requested-with'));

            if (($requestWith == 'xmlhttprequest') || (strpos($requestWith, 'shockwaveflash') !== false)) {
                $this->_isAjax = true;
            } else {
                $this->_isAjax = false;
            }
        }

        return $this->_isAjax;
    }
```

This patch has been applied on **version 0.1.11** just released.

**Comment by LucaGallinari** [comment]
I still get this problem in Magento 1.9.3.2. This is probably due to changes made to the image uploader of 1.9.3.0, they removed the flash uploader.

**Comment by phoenix128** [comment]
Uhm... seems more a Magento issue.
Magento is giving back `Content-Type:text/html` instead of `application/json`.

We are looking for a way to fix it.

**Comment by phoenix128** [comment]
Fixed in 0.1.17
